python/test/autoglome_test.py

```diff
@@ -0,0 +1,116 @@
+        self.sender_glome = pyglome.AutoGlome(peer_key, my_key,
+                                              truncated_tag_length,
+                                              skippable_range)
+
+        peer_key = x25519.X25519PublicKey.from_public_bytes(self.data.ka)
+        my_key = x25519.X25519PrivateKey.from_private_bytes(self.data.kbp)
+        self.receiver_glome = pyglome.AutoGlome(peer_key, my_key,
+                                                truncated_tag_length,
+                                                skippable_range)
+
+
+class AutoGlomeTestBase:
+    """Test keys constructions, tag generation and tag checking for AutoGlome."""
+
+    def __init__(self, *args, **kwargs):
+        self.test_vector = None
+
+    def test_check_counters_raise_exceptions_when_incorrect(self):
+        test_vector = self.test_vector
+        with self.assertRaises(ValueError):
+            test_vector.sender_glome.sending_counter = -1
+        with self.assertRaises(ValueError):
+            test_vector.receiver_glome.sending_counter = 256
+        with self.assertRaises(ValueError):
+            test_vector.sender_glome.receiving_counter = 280
+        with self.assertRaises(ValueError):
+            test_vector.receiver_glome.receiving_counter = 280
+
+    def test_check_counters_dont_raise_exceptions_when_correct(self):
+        test_vector = self.test_vector
+        try:
+            test_vector.sender_glome.sending_counter = 0
+            test_vector.receiver_glome.sending_counter = 23
+            test_vector.sender_glome.receiving_counter = 123
+            test_vector.receiver_glome.receiving_counter = 255
+        except ValueError:
+            self.fail('properties raised ValueError unexpectedly!')
+
+    def test_check_counters_are_correctly_setted(self):
+        test_vector = self.test_vector
+        test_vector.sender_glome.sending_counter = 0
+        self.assertEqual(test_vector.sender_glome.sending_counter, 0)
+        test_vector.receiver_glome.sending_counter = 23
+        self.assertEqual(test_vector.receiver_glome.sending_counter, 23)
+        test_vector.sender_glome.receiving_counter = 123
+        self.assertEqual(test_vector.sender_glome.receiving_counter, 123)
+        test_vector.receiver_glome.receiving_counter = 255
+        self.assertEqual(test_vector.receiver_glome.receiving_counter, 255)
+
+    def test_tag(self):
+        test_vector = self.test_vector
+        test_vector.sender_glome.sending_counter = test_vector.data.counter
+        self.assertEqual(test_vector.sender_glome.tag(test_vector.data.msg),
+                         test_vector.data.tag)
+
+    def test_skippable_range(self):
+        test_vector = self.test_vector
+        try:
+            test_vector.receiver_glome.receiving_counter = test_vector.data.counter
+            test_vector.receiver_glome.check(test_vector.data.tag,
+                                             msg=test_vector.data.msg)
```

Comment:
I don't quite understand how this tests the skippable range, could you maybe describe what the test does?

Reply:
It was wrong, and wasn't doing what I intended. It has been updated so now:
- set the receiving counter to counter-skippable range
- ensures the message is correctly checked
- assert the counter is correctly setted afterwards